app/routes/questionnaire.py

```diff
@@ -253,23 +279,23 @@ def relationship(schema, questionnaire_store, block_id, list_item_id, to_list_it
     except InvalidLocationException:
         return redirect(url_for('.get_questionnaire'))
 
-    form = _generate_wtf_form(
+    block_handler.form = _generate_wtf_form(
```

Comment:
It made sense for the form to be part of the block handler as it is used pretty heavily within block handler. Going forward we should look at constructing the form on initialisation of the block handler. That was not done as part of this PR to keep it simple as block handler would need `request` object to do this.